### PR TITLE
refactor(e2e): Address gap in registry manager

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Bulk device and twin APIs were missing an e2e test

### CODE REVIEW

File: e2e/test/iothub/service/RegistryManagerE2ETests.cs [hunk 1]
```diff
@@ -87,6 +88,62 @@ public async Task RegistryManager_AddDeviceWithTwinWithDeviceCapabilities()
             }
         }
 
+        [LoggedTestMethod]
+        public async Task RegistryManager_BulkLifecycle()
+        {
+            int bulkCount = 50;
+            string[] deviceIds = new string[bulkCount];
```
Comment (vinagesh) [inline e2e/test/iothub/service/RegistryManagerE2ETests.cs:95]: This is not really needed. You can just iterate over devices and get the deviceId from that in the for loop below.

Comment (timtay-microsoft) [reply e2e/test/iothub/service/RegistryManagerE2ETests.cs:95]: True

File: e2e/test/iothub/service/RegistryManagerE2ETests.cs [hunk 2]
```diff
@@ -87,6 +88,62 @@ public async Task RegistryManager_AddDeviceWithTwinWithDeviceCapabilities()
...
+            var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+
+            // Test that you can create devices in bulk
+            var bulkAddResult = await registryManager.AddDevices2Async(devices).ConfigureAwait(false);
+            Assert.IsTrue(bulkAddResult.IsSuccessful);
+
+            foreach (Device device in devices)
+            {
+                // After a bulk add, every device should be able to be retrieved
+                Assert.IsNotNull(await registryManager.GetDeviceAsync(device.Id).ConfigureAwait(false));
```
Comment (vinagesh) [inline e2e/test/iothub/service/RegistryManagerE2ETests.cs:112]: This is totally fine but if we are following fluent assertions in general, we should be consistent and use it. Ex: .Should().Be()

Comment (timtay-microsoft) [reply e2e/test/iothub/service/RegistryManagerE2ETests.cs:112]: We don't really use that pattern in this repo for whatever reason. I'd rather keep the pattern here

File: e2e/test/iothub/service/RegistryManagerE2ETests.cs [hunk 3]
```diff
@@ -87,6 +88,62 @@ public async Task RegistryManager_AddDeviceWithTwinWithDeviceCapabilities()
...
+
+            // Test that you can delete device identities in bulk
+            var bulkDeleteResult = await registryManager.RemoveDevices2Async(devices, true, default).ConfigureAwait(false);
+
+            Assert.IsTrue(bulkDeleteResult.IsSuccessful);
+
+            foreach (Device device in devices)
+            {
+                // After a bulk delete, every device should not be found
+                Assert.IsNull(await registryManager.GetDeviceAsync(device.Id).ConfigureAwait(false));
```
Comment (vinagesh) [inline e2e/test/iothub/service/RegistryManagerE2ETests.cs:143]: Will this not throw 404?

Comment (timtay-microsoft) [reply e2e/test/iothub/service/RegistryManagerE2ETests.cs:143]: Surprisingly, it doesn't. I was expecting it to, but it just returns null

Comment (timtay-microsoft) [reply e2e/test/iothub/service/RegistryManagerE2ETests.cs:143]: I'm not looking to change the behavior of the API at this point, though

Comment (vinagesh) [reply e2e/test/iothub/service/RegistryManagerE2ETests.cs:143]: Ohh really? That is surprising to me :) Ofcourse we cannot change the behavior of the API but I was expecting a 404.

**Pull request merged**